ivy/functional/backends/paddle/elementwise.py

```diff
@@ -494,15 +342,11 @@ def divide(
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
     x1, x2, ret_dtype = _elementwise_helper(x1, x2)
-    if x1.dtype in [paddle.float16, paddle.bfloat16]:
-        x1, x2 = x1.astype("float32"), x2.astype("float32")
-    if not (ivy.is_float_dtype(ret_dtype) or ivy.is_complex_dtype(ret_dtype)):
-        ret_dtype = ivy.default_float_dtype(as_native=True)
     return (x1 / x2).astype(ret_dtype)
 
 
 @with_supported_dtypes(
-    {"2.5.1 and below": ("float64", "float32", "int64", "int64")},
+    {"2.5.1 and below": ("float", "int32", "int64")},
```

Comment:
`int32` wasn't originally present in the decorator, have you tested if it works for this function?